Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/ContentDialogSamplePage.xaml

```diff
@@ -12,14 +12,97 @@
 	<Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
 		<local:SamplePageLayout>
 
+			<local:SamplePageLayout.MaterialTemplate>
+				<DataTemplate>
+					<StackPanel>
+						<smtx:XamlDisplay UniqueKey="ContentDialogSamplePage_Material_Default"
+										  smtx:XamlDisplayExtensions.Header="Default ContentDialog">
+
+							<Button Content="Open dialog"
+									Click="Button_Click" />
+							<!-- with the following C# code
+private async void Button_Click(object sender, RoutedEventArgs e)
+{
+	var contentDialog = new ContentDialog()
+	{
+		Content = "Hello world!",
+		PrimaryButtonText = "OK",
+		XamlRoot = this.XamlRoot
+	};
+	await contentDialog.ShowAsync();
+}
+-->
+						</smtx:XamlDisplay>
+
+						<smtx:XamlDisplay UniqueKey="ContentDialogSamplePage_Material_Title"
+										  smtx:XamlDisplayExtensions.Header="ContentDialog with Title">
+
+							<Button Content="Open dialog"
+									Click="Button_Click2" />
+							<!-- with the following C# code
+private async void Button_Click2(object sender, RoutedEventArgs e)
+{
+	var contentDialog = new ContentDialog
+	{
+		Title = "Notice",
+		Content = "This is a very important message.",
+		PrimaryButtonText = "OK",
+		XamlRoot = this.XamlRoot
+	};
+	await contentDialog.ShowAsync();
+}
+-->
+						</smtx:XamlDisplay>
+
+						<smtx:XamlDisplay UniqueKey="ContentDialogSamplePage_Material_Prompt"
+										  smtx:XamlDisplayExtensions.Header="ContentDialog with Prompt">
+							<Button Content="Open dialog"
+									Click="Button_Click3" />
+							<!-- with the following C# code
+private async void Button_Click3(object sender, RoutedEventArgs e)
+{
+	var contentDialog = new ContentDialog()
+	{
+		Title = "Log out",
+		Content = "Are you sure you want to log out?",
+		PrimaryButtonText = "Log out",
+		CloseButtonText = "Cancel",
+		DefaultButton = ContentDialogButton.Primary,
+		XamlRoot = this.XamlRoot,
+	};
+
+	if (await contentDialog.ShowAsync() == ContentDialogResult.Primary)
+	{
+		// Log out
+	}
+	else
+	{
+		// Cancel
+	}
+}
+-->
+						</smtx:XamlDisplay>
+					</StackPanel>
+				</DataTemplate>
+			</local:SamplePageLayout.MaterialTemplate>
+
 			<local:SamplePageLayout.FluentTemplate>
 				<DataTemplate>
 					<StackPanel>
+						<StackPanel.Resources>
+							<ResourceDictionary>
+								<ResourceDictionary.MergedDictionaries>
+									<XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
+								</ResourceDictionary.MergedDictionaries>
+							</ResourceDictionary>
+						</StackPanel.Resources>
+						
 						<smtx:XamlDisplay UniqueKey="ContentDialogSamplePage_Fluent_Default"
 										  smtx:XamlDisplayExtensions.Header="Default ContentDialog">
 
 							<Button Content="Open dialog"
-									Click="Button_Click" />
+									Click="Button_Click"
+									Tag="Fluent" />
```

Comment:
dont add `Tag="Fluent"` here
viewers may think this is some sort of witchcraft to make this work
(in fact, it is... but specific to our repo due to competing libraries)
but they shouldn't know that...

Reply:
see my comment in the corresponding cs file, for alternative

Reply:
Thanks Xiaoy312, with some minor adjustments from my side your suggestion works well.